pkg/kubernetes/api/core/v1/pod/build.go

```diff
@@ -90,6 +91,34 @@ func (b *Builder) WithNamespace(namespace string) *Builder {
 	return b
 }
 
+// WithLabels sets the labels of Pod with provided value.
+func (b *Builder) WithLabels(matchlabels map[string]string) *Builder {
+	if len(matchlabels) == 0 {
+		b.errs = append(
+			b.errs,
+			errors.New("failed to build deployment object: no new matchlabels"),
+		)
+		return b
+	}
+
+	// copy of original map
+	newmatchlabels := map[string]string{}
+	for key, value := range matchlabels {
+		newmatchlabels[key] = value
+	}
+
```

Comment:
For appending we also need to add below code 
```suggestion
for key, value := range b.pod.object.Labels {
    newmatchlabels[key] = value
}
```

Reply:
Ahh!! missed it.

Reply:
you don't need to use an additional map. you can directly add to the existing pod label..